src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs

```diff
@@ -1674,12 +2880,265 @@ public class Bar
     public void Sub(IEnumerable<int> i)
     {
         IEnumerable<int> j = Enumerable.Range(1, 10).ToArray();
-        var z = i.Concat(j);
+        var z = i.Zip(j, (a, b) => (a, b));
         j.ElementAt(10);
         z.ToArray();
     }
 }";
-            await VerifyCS.VerifyAnalyzerAsync(code);
+            await VerifyCS.VerifyAnalyzerAsync(csharpCode);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace NS
+    Public Class Bar
+        Public Sub Goo(i As IEnumerable(Of Integer))
+            Dim j As IEnumerable(Of Integer) = Enumerable.Range(1, 10).ToArray()
+            Dim z = i.Zip(j, Function(a, b) (a, b))
+            j.ElementAt(10)
+            z.ToArray()
+        End Sub
+    End Class
+End Namespace";
+
+            await VerifyVB.VerifyAnalyzerAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestCallExtensionMethodAsOrdinaryMethod()
```

Comment:
Starting from here is the 'real' new tests added in this PR.
All the previous change in this tests file are just adding the corresponding VB tests